Bukkit/src/main/java/com/github/intellectualsites/plotsquared/bukkit/listeners/PlayerEvents.java

```diff
@@ -776,7 +776,7 @@ public void onTeleport(PlayerTeleportEvent event) {
                 // to is identical to the plot's home location, and untrusted-visit is true
                 // i.e. untrusted-visit can override deny-teleport
                 // this is acceptable, because otherwise it wouldn't make sense to have both flags set
-                if (!result && !(plot.getFlag(UntrustedVisitFlag.class) && plot.getHome()
+                if (!result && !(plot.getFlag(UntrustedVisitFlag.class) && plot.getHomeSynchronous()
```

Comment:
Because we need to be able to cancel it, and if we trigger an asynchronous chunk load then we won't be able to cancel the event in the case that the equality check fails.